v-next/hardhat/src/internal/cli/main.ts

```diff
@@ -45,6 +45,10 @@ export interface MainOptions {
   rethrowErrors?: true;
 }
 
+declare global {
```

Comment:
What if instead we modify the `MainOptions` a bit, so that it only defaults `registerTsx` to true if it's running on Node.js? That way we don't need any `declare` nor dependency.

I believe we can detect node like this:

```ts
const isNode = typeof process === 'object' &&
               process?.versions?.node !== undefined;
```

Reply:
We could also default it to false, and do the `isNode` evaluation in `src/index.ts`.

Reply:
Checking if something "isNode" is unfortunately hard

The reason is that a lot of existing code has some equivalent of "isNode" to differentiate backend vs frontend. To make sure those projects still work in different runtimes like Deno and Bun, both of the return `true` for this kind of isNode process despite the fact they aren't actually nodejs

Reply:
I made the type `Deno: any` specifically to avoid having to bring in any dependency so that it's just type stuff that disappears when you actually compile

Reply:
> To make sure those projects still work in different runtimes like Deno and Bun, both of the return true for this kind of isNode process despite the fact they aren't actually nodejs

Oh, I was scared that would be the situation :( but is understandable.

I guess we should go for a blocklist approach, and not register it under runtimes we know it's not needed.

Then maybe we can detect Deno without any type declaration with `"Deno" in globalThis`.